### DISCUSSION TITLE
What Problem is Transition Texture between RenderPass in an CommandEncoder

### DISCUSSION THREAD

**Discussion by ghost** [gfx-rs/wgpu#4135] [General]
Is it possible to use the same texture for different purposes within multiple RenderPasses under a single CommandEncoder? For example, can the texture serve as a render target in the first RenderPass and then be used for shader sampling in the second RenderPass? 

Here's a simplified example:

``` rust

let t1 = some_texture;
{
    let rp1 = encoder.begin_render_pass(..., using t1 as render target);
    // ...
}
let t2 = another_texture;
{
    let rp2 = encoder.begin_render_pass(..., using t2);
    // ...
    rp2.set_bind_group(0, bind_group_that_uses_t1);
    // ...
}
```

If this is not possible, are there specific limitations imposed by platform-specific APIs? 

What does the WebGPU standard say about this?

**Comment by kpreid** [comment]
> Is it possible to use the same texture for different purposes within multiple RenderPasses under a single CommandEncoder?

Yes, I do just that for my bloom implementation. Not only that, I'm simultaneously using different mip levels of the same texture.

Caveat: It doesn't work correctly under the `webgl` backend; this is very likely a bug or an unnoticed downlevel limitation, but I haven't gotten to preparing a proper bug report. (The symptom is nothing being rendered and a console error `Feedback loop formed between Framebuffer and active Texture.` if on Chrome.)

**Reply by ghost** [reply]
thanks very much, I check my error again.
